src/modules/blockchainApplication/connection/components/RequestSummary/index.js

```diff
@@ -139,6 +142,13 @@ const RequestSummary = ({ nextStep, history, message }) => {
           validator.validator.validate(schema, transactionObj.params);
           setTransaction(transactionObj);
         }
+        if (currentApplication.chainID !== sendingChainID) {
```

Comment:
Feels like this is the first type of error we should check for, when we get a request. Not after doing validations etc. 

It could be moved to top execution for this components lifecycle, which is in the useEffect above this one. 

```
  useEffect(() => {
    const event = events.find((e) => e.name === EVENTS.SESSION_REQUEST);
    const hasRequest = event?.meta?.params?.request?.params;

    if (hasRequest) {
      const eventChainId = event?.meta?.params?.chainId?.replace('lisk:', ''); // Something like this?

      if (currentApplication.chainID !== eventChainId) {
        setErrorMessage(
          t('Please switch application/network to selected value during connection initialization')
        );
      } else {
        setRequest({
          id: event.meta.id,
          ...event.meta.params,
        });
      }
    }
  }, []);
```

And then make sure we dont show emptyState if we have an error message 🤔

Reply:
Fixed